### PR TITLE
[Cart] Extract and validate change item quantity action

### PR DESCRIPTION
**Pull request by lchrusciel**
Based on #142

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Requires rebasing.

File: spec/Command/ChangeItemQuantitySpec.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        $this->itemIdentifier()->shouldReturn(1);
+    }
+
+    function it_has_quantity()
+    {
+        $this->quantity()->shouldReturn(5);
+    }
+
+    function it_throws_an_exception_if_order_token_is_not_a_string()
```
Comment (pamil) [inline spec/Command/ChangeItemQuantitySpec.php]: Can't we use PHP 7.0 typehints?

File: spec/Handler/ChangeItemQuantityHandlerSpec.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        $this->beConstructedWith($orderItemRepository, $orderRepository, $orderItemModifier, $orderProcessor);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(ChangeItemQuantityHandler::class);
+    }
+
+    function it_handles_putting_new_item_to_cart(
+        OrderInterface $order,
```
Comment (pamil) [inline spec/Handler/ChangeItemQuantityHandlerSpec.php]: Why don't we keep the `let` order of arguments?

File: spec/Command/ChangeItemQuantitySpec.php (hunk 1 above)
Comment (lchrusciel) [inline spec/Command/ChangeItemQuantitySpec.php]: This code has been written before support for php 70 was introduced. Can be done in the next iteration

**Comment by pamil** [comment]
👍

**Pull request merged**